sch_yang_condition.c

```diff
@@ -0,0 +1,923 @@
+        case XPATH_TYPE_LT:
+            if (is_number)
+                presult->result = lnumber < rnumber;
+            break;
+        case XPATH_TYPE_LE:
+            if (is_number)
+                presult->result = lnumber <= rnumber;
+            break;
+        case XPATH_TYPE_GT:
+            if (is_number)
+                presult->result = lnumber > rnumber;
+            break;
+        case XPATH_TYPE_GE:
+            if (is_number)
+                presult->result = lnumber >= rnumber;
+            break;
+        case XPATH_TYPE_PLUS:
+            if (is_number)
+            {
+                presult->value = g_strdup_printf ("%d", lnumber + rnumber);
+                presult->result = true;
+            }
+            break;
+        case XPATH_TYPE_MINUS:
+            if (is_number)
+            {
+                presult->value = g_strdup_printf ("%d", lnumber - rnumber);
+                presult->result = true;
+            }
+            break;
+        case XPATH_TYPE_MULTIPLY:
+            if (is_number)
+            {
+                presult->value = g_strdup_printf ("%d", lnumber * rnumber);
+                presult->result = true;
+            }
+            break;
+        case XPATH_TYPE_DIVIDE:
+            if (is_number && rnumber)
+            {
+                presult->value = g_strdup_printf ("%d", lnumber / rnumber);
+                presult->result = true;
+            }
+            break;
+        case XPATH_TYPE_MODULO:
+            if (is_number && rnumber)
+            {
+                presult->value = g_strdup_printf ("%d", lnumber % rnumber);
+                presult->result = true;
+            }
+            break;
+        case XPATH_TYPE_UNARY_MINUS:
+            if (is_number)
+            {
+                presult->value = g_strdup_printf ("-%d", lnumber);
+                presult->result = true;
+            }
+            break;
+        default:
+            presult->result = false;
+            break;
+        }
+    }
+    g_free (lresult.value);
+    g_free (rresult.value);
+}
+
+static void
+sch_yang_process_predicate (sch_instance *instance, GNode *root, char *path,
+                            char *step_path, xpath_node *xnode, yang_result *presult,
+                            int depth, int *flags)
+{
+    yang_result lresult = { };
+    yang_result rresult = { };
+
+    presult->result = false;
+    if (xnode->left)
+    {
+        sch_yang_process_xnode (instance, root, path, step_path, xnode->left, &lresult,
+                                depth + 1, flags);
+        presult->result = lresult.result;
+        if (lresult.result && xnode->right)
+        {
+            sch_yang_process_xnode (instance, root, path, lresult.value, xnode->right,
+                                    &rresult, depth + 1, flags);
+            presult->result = rresult.result;
+        }
+    }
+    if (rresult.result)
+    {
+        presult->value = rresult.value;
+        rresult.value = NULL;
```

Comment:
could free lresult here and not clear rresult

Reply:
Done